Open and closed events

frimpongopoku opened the pull request:
### Works with API PR https://github.com/massenergize/api/pull/606
### Works with frontend PR https://github.com/massenergize/frontend-portal/pull/1128

- [X] Admins can set event coverage ( Open, closed, open to, or closed to) 

https://user-images.githubusercontent.com/26961591/199160153-8183da7d-20e4-49d5-b95c-623dfe2065e7.mov


https://user-images.githubusercontent.com/26961591/199161791-71fd46a2-e1fb-4d60-adee-4251488ba40a.mov


- [X] CAdmins events  page is divided into two now. One tab for the communities they manage, and the other tab for other communities. There are no conflicts this way, and the code is tidier. 
<img width="1477" alt="Screenshot 2022-11-01 at 08 16 25" src="https://user-images.githubusercontent.com/26961591/199160295-4932b0bc-d73e-4466-b3a7-b3f62db1bd9a.png">

- [X] Events can be viewed in full. Whether its one from their community, or other people's community 

 
<img width="1475" alt="Screenshot 2022-11-01 at 08 17 19" src="https://user-images.githubusercontent.com/26961591/199160348-7f4f3689-72a2-4f58-b67d-6905e5545af1.png">

- [X]  Share button will only be activated if the event is **open**, or any of the  admin's communities have been listed in the **Open_to** list 

- [X] The edit button will also only be activated only if the event is in their admins community and so on. 

- [X] Share button opens up a dialog that will only list the communities that an event can be shared to based on an admin's community list. For super admins, its every community when its open, or all the allowed communities when its open_to
<img width="1310" alt="Screenshot 2022-11-01 at 07 58 07" src="https://user-images.githubusercontent.com/26961591/199160843-c2680554-5bbb-4879-af94-50fb1d62cc63.png">

### SHARING 


https://user-images.githubusercontent.com/26961591/199160986-ed778007-8bce-4709-a039-4d1d25c9639b.mov
### FOR CADMINS 

<img width="1473" alt="Screenshot 2022-11-01 at 08 10 12" src="https://user-images.githubusercontent.com/26961591/199161204-b6a08043-554b-4b83-bdc4-3f9e0ec1e600.png">


### FETCHING EVENTS FROM OTHER COMMUNITIES 


https://user-images.githubusercontent.com/26961591/199161311-f67a6eca-655e-4f57-a3cb-6b3356c98bde.mov

frimpongopoku commented:
@abdullai-t   this is the new link that you should put in the nudge email 

**Sharing :** `http://localhost:3001/admin/read/event/<event-id-here>/event-view?from=others&dialog=open`** 
**Just View:** `http://localhost:3001/admin/read/event/<event-id-here>/event-view?from=main`** 
If they are signed in,  it will take them to the full view page. If not they will be asked to sign in, and then taken back

Davidjwkinghorn commented:
Hey @frimpongopoku, here's some edits from Brad, Kaat, Aimee, and myself.
https://docs.google.com/document/d/10vUZkZK0uxRHNAabpktP8gIppWNmxODjA71Yl18tKJI/edit#

frimpongopoku commented:
- [X] Changed "event publicity" 
- [X] Removed "Exclude certain communities" use case 
- [X] Removed tabs 
- [X] Made tags filter a dropdown 
- [X] Changed "fetch" to "show events" 
- [X] Removed "copy" from full view page 
- [X] Changed "start date"  to "date" 
- [X] Removed "end date"  
- [X] Made full event view into an iframe

Davidjwkinghorn commented:
We should have two separate tabs on the sidebar. One for just the events in the cadmin's communities and another for all events from other communities. We want to keep these views separate as they're two different use cases.

![Screen Shot 2022-11-16 at 4 22 54 PM](https://user-images.githubusercontent.com/15150372/203421318-9367e681-b3e7-425a-a9ec-6ef559e5f98b.png)

frimpongopoku commented:
> We should have two separate tabs on the sidebar. One for just the events in the cadmin's communities and another for all events from other communities. We want to keep these views separate as they're two different use cases.
> 
> ![Screen Shot 2022-11-16 at 4 22 54 PM](https://user-images.githubusercontent.com/15150372/203421318-9367e681-b3e7-425a-a9ec-6ef559e5f98b.png)

@Davidjwkinghorn  I think you need to pull in the new changes, I added that already. 🤔

Davidjwkinghorn commented:
Yes that's what was specified in the feedback doc. I have been running the most recent commits on this branch. I think it was a caching issue on my end. It's showing up properly now.

Davidjwkinghorn reviewed (approved):
LGTM
